Missing support for some new SDL2 gamepad types

```
Source/controls/devices/game_controller.cpp:246:16: warning: enumeration value ‘SDL_CONTROLLER_TYPE_NVIDIA_SHIELD’ not handled in switch [-Wswitch]
  246 |         switch (gamepadType) {
      |                ^
Source/controls/devices/game_controller.cpp:246:16: warning: enumeration value ‘SDL_CONTROLLER_TYPE_NINTENDO_SWITCH_JOYCON_LEFT’ not handled in switch [-Wswitch]
Source/controls/devices/game_controller.cpp:246:16: warning: enumeration value ‘SDL_CONTROLLER_TYPE_NINTENDO_SWITCH_JOYCON_RIGHT’ not handled in switch [-Wswitch]
Source/controls/devices/game_controller.cpp:246:16: warning: enumeration value ‘SDL_CONTROLLER_TYPE_NINTENDO_SWITCH_JOYCON_PAIR’ not handled in switch [-Wswitch]
```

We should figure out which SDL versions introduced them and set the appropriate gamepad type for these.

--- a/Source/controls/devices/game_controller.cpp
+++ b/Source/controls/devices/game_controller.cpp
@@ -245,6 +245,11 @@ GamepadLayout GameController::getLayout(const SDL_Event &event)
 	const SDL_GameControllerType gamepadType = SDL_GameControllerTypeForIndex(index);
 	switch (gamepadType) {
 	case SDL_CONTROLLER_TYPE_NINTENDO_SWITCH_PRO:
+#if SDL_VERSION_ATLEAST(2, 24, 0)
+	case SDL_CONTROLLER_TYPE_NINTENDO_SWITCH_JOYCON_LEFT:
+	case SDL_CONTROLLER_TYPE_NINTENDO_SWITCH_JOYCON_RIGHT:
+	case SDL_CONTROLLER_TYPE_NINTENDO_SWITCH_JOYCON_PAIR:
+#endif
 		return GamepadLayout::Nintendo;
 	case SDL_CONTROLLER_TYPE_PS3:
 	case SDL_CONTROLLER_TYPE_PS4:
@@ -257,6 +262,9 @@ GamepadLayout GameController::getLayout(const SDL_Event &event)
 #if SDL_VERSION_ATLEAST(2, 0, 16)
 	case SDL_CONTROLLER_TYPE_GOOGLE_STADIA:
 	case SDL_CONTROLLER_TYPE_AMAZON_LUNA:
+#if SDL_VERSION_ATLEAST(2, 24, 0)
+	case SDL_CONTROLLER_TYPE_NVIDIA_SHIELD:
+#endif
 #endif
 		return GamepadLayout::Xbox;
 #if SDL_VERSION_ATLEAST(2, 0, 14)

